acis_thermal_check/regression_testing.py

```diff
@@ -115,9 +122,13 @@ def __init__(self, name, outdir, model_path, run_start=None,
                         }
              }
 
-def exception_catcher(test, old, new, data_type):
+def exception_catcher(test, old, new, data_type, **kwargs):
+    if "S" in new.dtype.descr[-1][-1]:
```

Comment:
Is this supposed to be `old.dtype` not `new.dtype`?

Also, `new.dtype.kind` is probably what you want here.  It will be either `'S'` or `'U'`.